support-frontend/assets/helpers/forms/checkouts.ts

```diff
@@ -135,7 +136,7 @@ function getAmountFromUrl(): number | 'other' | null {
 		return 'other';
 	}
 
-	const amount = parseInt(selected, 10);
+	const amount = parseInt(selected as string, 10);
```

Comment:
I'm no way an expert on typescript, but does `as string` add a compile time type where it's `any`?  I'm just wondering whether `getQueryParameter` should return a `string` in the first place, if it doesn't already.  Any thoughts?

Reply:
I'm type casting the `selected` argument here because the the first parameter in `getQueryParameter `

The `parseInt` function only accepts a string so `as string` here will type cast the result of calling `getQueryParameter` to a string at compile time vs the given return types of the function i.e. `string | null | undefined`.

And looking at how the `getQueryParameter` function is used I agree we could include a default value for the parameter of an empty string and therefore the return type would be a string.

Reply:
ok I see so it's kind of like an optional string coming back, and we wanted to tell the type checker, actually we know it will be present.  Makes more sense in that case, but I think your suggestion of a default makes sense also because we get even more comprehensive type checking.